Sources/ElasticSwift/Requests/IndexRequest.swift

```diff
@@ -61,104 +67,101 @@ public class IndexRequestBuilder<T: Codable>: RequestBuilder {
         self.completionHandler = completionHandler
         return self
     }
     
-    public func build() throws -> Request {
-        return try IndexRequest(withBuilder: self)
+    public func make() throws -> IndexRequest<T> {
+        return try IndexRequest<T>(withBuilder: self)
     }
     
 }
 
 public class IndexRequest<T: Codable>: Request {
     
-    let client: ESClient
-    let completionHandler: ((_ response: IndexResponse?, _ error: Error?) -> Void)
-    public var method: HTTPMethod  {
-        get {
-            if self.id == nil {
-                return .POST
-            }
-            return .PUT
-        }
-    }
+    public typealias ResponseType = IndexResponse
+    
+    public let completionHandler: ((IndexResponse?,Error?) -> ())?
     
-    var _builtBody: Data?
-    var index: String?
+    public let client: ESClient
+    var index: String
     var type: String?
     var id: String?
-    var source: T?
+    var source: T
     var routing: String?
     var parent: String?
-    
+    var refresh: IndexRefresh?
     
     init(withBuilder builder: IndexRequestBuilder<T>) throws {
         self.client = builder.client
-        self.completionHandler = builder.completionHandler!
         self.index = builder.index
+        self.source = builder.source
+        
         self.type = builder.type
         self.id = builder.id
-        self.source = builder.source
         self.routing = builder.routing
         self.parent = builder.parent
-        self._builtBody = try makeBody()
-    }
-    
-    func makeEndPoint() -> String {
-        var _endPoint = self.index! + "/" + self.type!
-        if let id = self.id {
-            _endPoint = _endPoint + "/" + id
-        }
-        return _endPoint
+        
+        self.refresh = builder.refresh
+        self.completionHandler = builder.completionHandler
     }
     
-    public var endPoint: String {
+    public var method: HTTPMethod  {
         get {
-            return makeEndPoint()
+            if self.id == nil {
+                return .POST
+            }
+            return .PUT
         }
     }
     
-    public var body: Data {
+    public var endPoint: String {
         get {
-            return _builtBody!
+            var result = self.index + "/"
+            
+            if let type = self.type {
+                result += type + "/"
+            } else {
+                result += "_doc/"
+            }
+            
+            if let id = self.id {
+                result += id
+            }
+            
+            return result
         }
     }
     
-    public func makeBody() throws -> Data {
-       return try Serializers.encode(self.source!)!
-    }
-    
-    public func execute() {
-        self.client.execute(request: self, completionHandler: responseHandler)
+    public func getBody() throws -> Data? {
+        return try self.serializer.encode(self.source)
```

Comment:
Gereralized De/Serializers defined in
```
extension Request {
    public var serializer : Serializer {
        return DefaultSerializer()
    }
...
}
```

So you can customize i.e. DateTime formatters for specific Requests.
Current implementation in ```DefaultSerializer```